boards/msp_exp432p401r/src/main.rs

```diff
@@ -183,18 +218,75 @@ pub unsafe fn reset_handler() {
     // Create the debugger object that handles calls to `debug!()`.
     components::debug_writer::DebugWriterComponent::new(uart_mux).finalize(());
 
+    // Setup alarm
     let timer0 = &msp432::timer::TIMER_A0;
     let mux_alarm = components::alarm::AlarmMuxComponent::new(timer0).finalize(
         components::alarm_mux_component_helper!(msp432::timer::TimerA),
     );
     let alarm = components::alarm::AlarmDriverComponent::new(board_kernel, mux_alarm)
         .finalize(components::alarm_component_helper!(msp432::timer::TimerA));
 
+    // Setup ADC
+
+    setup_adc_pins();
+
+    let adc_channels = static_init!(
+        [&'static msp432::adc::Channel; 24],
+        [
+            &msp432::adc::Channel::Channel0,  // A0
+            &msp432::adc::Channel::Channel1,  // A1
+            &msp432::adc::Channel::Channel2,  // A2
+            &msp432::adc::Channel::Channel3,  // A3
+            &msp432::adc::Channel::Channel4,  // A4
+            &msp432::adc::Channel::Channel5,  // A5
+            &msp432::adc::Channel::Channel6,  // A6
+            &msp432::adc::Channel::Channel7,  // A7
+            &msp432::adc::Channel::Channel8,  // A8
+            &msp432::adc::Channel::Channel9,  // A9
+            &msp432::adc::Channel::Channel10, // A10
+            &msp432::adc::Channel::Channel11, // A11
+            &msp432::adc::Channel::Channel12, // A12
+            &msp432::adc::Channel::Channel13, // A13
+            &msp432::adc::Channel::Channel14, // A14
+            &msp432::adc::Channel::Channel15, // A15
+            &msp432::adc::Channel::Channel16, // A16
+            &msp432::adc::Channel::Channel17, // A17
+            &msp432::adc::Channel::Channel18, // A18
+            &msp432::adc::Channel::Channel19, // A19
+            &msp432::adc::Channel::Channel20, // A20
+            &msp432::adc::Channel::Channel21, // A21
+            &msp432::adc::Channel::Channel22, // A22
+            &msp432::adc::Channel::Channel23, // A23
+        ]
+    );
+
+    let grant_cap = create_capability!(capabilities::MemoryAllocationCapability);
+    let grant_adc = board_kernel.create_grant(&grant_cap);
+    let adc = static_init!(
+        capsules::adc::Adc<'static, msp432::adc::Adc>,
+        capsules::adc::Adc::new(
+            &msp432::adc::ADC,
+            grant_adc,
+            adc_channels,
+            &mut capsules::adc::ADC_BUFFER1,
+            &mut capsules::adc::ADC_BUFFER2,
+            &mut capsules::adc::ADC_BUFFER3
+        )
+    );
+
+    msp432::adc::ADC.set_client(adc);
```

Comment:
Shouldn't this use a board component?

Reply:
Actually I really don't know.
I copied this part from the main.rs of the hail- and stm32-discovery board.
Do you mean that the `adc` variable should be a board component?

Reply:
Yeah, like the alarm component for example.

Reply:
> Yeah, like the alarm component for example.

I think I don't exactly get what you mean. As far as I can tell it's already a board-component. In line 289 in the main.rs this `adc` variable is added to the `MspExp432P401R` struct.

Reply:
Alistair is referring to "components", a tool that Tock uses to ensure uniform initialization across boards. There is an ADC component in `boards/components/src/adc.rs` which is used by several of the STM boards.  However, I think those components only work to construct a virtual ADC, which does not appear to be what you are trying to do here, so I think it is fine if this does not use a component for now.

Reply:
Ok, then I'll leave it now as it is. Thanks for the explanation!